src/components/Microsoft.DotNet.UpgradeAssistant.MSBuild/ComponentIdentifier.cs

```diff
@@ -88,8 +88,7 @@ private static async ValueTask<ProjectComponents> GetGeneralProjectComponentsAsy
             var components = ProjectComponents.None;
 
             // Check transitive dependencies
-            var nugetReferences = await project.GetNuGetReferencesAsync(token).ConfigureAwait(false);
-            if (MSBuildConstants.WinRTPackages.Any(package => nugetReferences.IsTransitivelyAvailable(package)))
+            if (await MSBuildConstants.WinRTPackages.ToAsyncEnumerable().AnyAwaitAsync(package => project.NuGetReferences.IsTransitivelyAvailableAsync(package, token), cancellationToken: token).ConfigureAwait(false))
```

Comment:
can we extract this to make the intent more clear? Should I read this as "if any package is a WinRT package then..."

Reply:
clarification, to me the language syntax of AsyncEnumerable is still new enough that I'm over focusing on that instead of the business rule you're building.

Reply:
@mjrousos Yes, it's for the async linq support. @KSchlobohm I'll refactor so it's clearer